zookeeper-docs/src/main/resources/markdown/zookeeperAdmin.md

```diff
@@ -1461,6 +1461,33 @@ and [SASL authentication for ZooKeeper](https://cwiki.apache.org/confluence/disp
     localhost (not over the network) or over an encrypted
     connection.
 
+* *DigestAuthenticationProvider.digestAlg* :
+    (Java system property: **zookeeper.DigestAuthenticationProvider.digestAlg**)
+     **New in 3.7.0:**
+    Set ACL digest algorithm. The default value is: `SHA1` which will be deprecated in the future for security issues.
+    Set this property the same value in all the servers.
+
+    - How to support other more algorithms?
+        - modify the `java.security` configuration file under `$JAVA_HOME/jre/lib/security/java.security` by specifying:
+             `security.provider.<n>=<provider class name>`.
+
+             ```
+             For example:
+             set zookeeper.DigestAuthenticationProvider.digestAlg=RipeMD160
+             security.provider.3=org.bouncycastle.jce.provider.BouncyCastleProvider
+             ```
+
+        - copy the jar file to `$JAVA_HOME/jre/lib/ext/`.
```

Comment:
isn't it enough to add it to the classpath ?

Modifying the JRE is something that usually no one can do, because people have the JDK/JRE pre-installed or managed by some system and modifying the contents of the directory can be forbidden in many  environments.

Also how to deal with this stuff in a Docker image ?

Reply:
I think it isn't hard to make these global changes even in docker. But if you need to do this only for your java process, then the standard way is:
- create your own security property file where you add the single line with the new security provider
- use the `-Djava.security.properties=<path/to/my/custom/properties>` system property

see: https://dzone.com/articles/how-override-java-security

Still, I think if you use a modern JVM, then there is a very good chance, that the hash algorithms in the default providers will be good enough for you. And if you need something really fancy (like Bouncy Castle FIPS Provider for some regulatory reasons) then you will need to tweak your security properies anyway...